### PR TITLE
Switch to IPFS Nucleus

### PR DESCRIPTION
**Pull request by ianopolous**

### CODE REVIEW

File: src/peergos/server/Main.java [hunk 1]
```diff
@@ -581,9 +581,9 @@ public static UserService startPeergos(Args a) {
                     a.getBoolean("include-csp", true), basicAuth, webroot, useWebAssetCache, isPublicServer, maxConnectionQueue, handlerThreads);
             boolean isPkiNode = nodeId.equals(pkiServerNodeId);
             if (! isPkiNode && useIPFS) {
-                int pkiNodeSwarmPort = a.getInt("pki.node.swarm.port");
-                InetAddress pkiNodeIpAddress = InetAddress.getByName(a.getArg("pki.node.ipaddress"));
-                ipfsWrapper.connectToNode(new InetSocketAddress(pkiNodeIpAddress, pkiNodeSwarmPort), pkiServerNodeId);
+//                int pkiNodeSwarmPort = a.getInt("pki.node.swarm.port");
```
Comment (kevodwyer) [inline src/peergos/server/Main.java:584]: maybe add a note to explain the comment

File: src/peergos/shared/storage/ContentAddressedStorage.java [hunk 2]
```diff
@@ -593,21 +593,17 @@ private static String encode(String component) {
 
         @Override
         public CompletableFuture<List<Multihash>> recursivePin(PublicKeyHash owner, Multihash hash) {
-            return poster.get(apiPrefix + PIN_ADD + "?stream-channels=true&arg=" + hash.toString()
-                    + "&owner=" + encode(owner.toString())).thenApply(this::getPins);
+            return Futures.of(Arrays.asList(hash));
```
Comment (kevodwyer) [inline src/peergos/shared/storage/ContentAddressedStorage.java:596]: can you explain these changes?

Comment (ianopolous) [reply src/peergos/shared/storage/ContentAddressedStorage.java:596]: We're not calling pin on ipfs at all anymore. This was only relevant when we were using the built-in ipfs GC, but we haven't been doing that for a while now. Pinning stops data from being garbage collected by the ipfs GC.

File: src/peergos/server/Main.java (hunk 1 above)
Comment (ianopolous) [reply src/peergos/server/Main.java:585]: done

**Review by kevodwyer — APPROVED** [review]

**Pull request merged**